packages/rum-recorder/src/domain/deflateWorker.d.ts

```diff
@@ -0,0 +1,32 @@
+export function createDeflateWorker(): DeflateWorker
+
+export interface DeflateWorker {
+  addEventListener(name: 'message', listener: DeflateWorkerListener): void
+  removeEventListener(name: 'message', listener: DeflateWorkerListener): void
+  postMessage(message: DeflateWorkerAction): void
+  terminate(): void
+}
+
+export type DeflateWorkerListener = (event: { data: DeflateWorkerResponse }) => void
+
+export type DeflateWorkerAction =
+  | {
+      id: number
+      action: 'write'
+      data: string
+    }
+  | {
+      id: number
+      action: 'complete'
+      data?: string
```

Comment:
I think we could, but since we don't need the final 'size' response. It would make things a bit more complex because it would loop in the case of a renew with max_size scenario:
1. we get a size response, with a size is bigger than the limit
2. we complete the segment and create a new segment
3. we get a size response, which is still bigger than the limit
4. we complete the segment (again). Because the new segment is empty, this would be a noop, but still looks weird.